src/compo/mopitt_co_nc2ioda.py

```diff
@@ -151,7 +152,15 @@ def _read(self):
             ap_tc = ap_tc.astype('float32')
 
             # set flag: rule out all anomalous data
-            flg = qa == 0
+            qaf = qa == 0
+
+            # date range to fit DA window
+            date_start = datetime.strptime(self.date_range[0], "%Y%m%d%H")
+            date_end = datetime.strptime(self.date_range[1], "%Y%m%d%H")
+            date_list = [datetime.strptime(date, "%Y-%m-%dT%H:%M:%SZ") for date in times]
+            tsf = [(date_i > date_start) & (date_i < date_end) for date_i in date_list]
```

Comment:
```suggestion
            tsf = [(date_i >= date_start) & (date_i < date_end) for date_i in date_list]
```

Reply:
@jeromebarre just need to fix this for if the date is exactly in the window, otherwise looks good